DEV-INTRO.md

```diff
@@ -1,59 +1,76 @@
-PEcAn Development
-=================
+# PEcAn Development
 
-Directory Structure
--------------------
+This is a minimal guide to getting started with PEcAn development.
 
-### pecan/
+## Git Repository and Workflow
 
-* modules/  Contains the modules that make up PEcAn
-* web/		The PEcAn web app to start a run and view outputs.
-* models/		Code to create the model specific configurations.
-* documentation/	Documentation about what PEcAn is and how to use it.
+The recommended workflow is gitflow which is described in the PEcAn documentation. For the repository we recommend using a [fork of the PEcAn repsitory](https://help.github.com/en/github/getting-started-with-github/fork-a-repo). In the scripts folder there is a script called [syngit.sh](scripts/syncgit.sh) that will help with synchronizing your fork with the official repository.
 
-### Modules (R packages)
+## Developing in Docker
 
-* General
-** all
-** utils
-** db
-* Analysis
-** modules/meta.analysis 
-** modules/uncertainty"
-** modules/data.land 
-** modules/data.atmosphere
-** modules/assim.batch 
-** modules/assim.sequential 
-** modules/priors
-* Model Interfaces
-** models/ed 
-** models/sipnet 
-** models/biocro
+To get started with development in docker we need to bring up the docker stack first. In the main pecan folder you will find the [docker-compose.yml](docker-compose.yml) file that can be used to bring up the pecan stack. There is also the [docker-compose.dev.yaml](docker-compose.dev.yaml) file that adds additional containers, and changes some services to make it easier for development.
 
+You can copy the `docker-compose.dev.yaml` to `docker-compose.override.yml`. Once that is done the `docker-compose` program will automatically use the `docker-compose.yml` and `docker-compose.override.yml` files, and you don't have to explicitly load them in the commands below.
 
-#### List of modules 
+### First time setup
 
-Installing PEcAn
-----------------
+You can copy the [`env.example`](docker/env.example) file as .env in your pecan folder. The variables we want to modify are:
 
-### Virtual Machine
+* `COMPOSE_PROJECT_NAME` set this to pecan, the prefix for all containers
+* `PECAN_VERSION` set this to develop, the docker image we start with
 
-* Fastest way to get started
-* see PEcAn demo ...
+Next we will create the folders that will hold all the data for the docker containers using: `mkdir -p volumes/{lib,pecan,portainer,postgres,rabbitmq,traefik}`. The `volumes` folder will be ignored by git.
 
-### Installing from source
+First we bring up postgresql (we will start RabbitMQ as well since it takes some time to start): `docker-compose -f docker-compose.yml -f docker-compose.dev.yml up -d postgres rabbitmq`. This will start postgresql and rabbitmq. We need to wait for a few minutes (you can look at the logs using `docker-compose logs postgres`) to see if it is ready.
```

Comment:
It is a bit confusing to use `docker-compose.dev.yml` here, but suggesting to copying it to  `docker.compose.override.yml` above. Although, I feel like for first time users this explicit command might be more helpful

Reply:
I get the following error from this computer:

```
ERROR: Configuration for volume traefik specifies "o" driver_opt bind, but a volume with the same name uses a different "o" driver_opt (None). If you wish to use the new configuration, please remove the existing volume "pecan_traefik" first:
$ docker volume rm pecan_traefik
```

Reply:
At this step I get errors when volumes (even stopped) exist

![image](https://user-images.githubusercontent.com/464871/81961768-af484100-95c7-11ea-95b1-ab1609ba63ae.png)

Reply:
looks like you have already the volumes called pecan_X, you can either remove them (`docker volume rm <name>`) or use a different prefix.

 **If you remove the volumes, all data will be lost!!**

To use a different prefix, change the `COMPOSE_PROJECT_NAME` in your .env